### PR TITLE
[mypyc] Implement `async for` as a statement and in comprehensions

### PR DESCRIPTION
**Pull request by msullivan**
Progress on mypyc/mypyc#868.

### CODE REVIEW

File: mypyc/primitives/generic_ops.py [hunk 1]
```diff
@@ -334,3 +334,19 @@
...
+# this would be aiter(obj) if it existed
+aiter_op = custom_op(
+    arg_types=[object_rprimitive],
+    return_type=object_rprimitive,
+    c_function_name="CPy_GetAIter",
+    error_kind=ERR_MAGIC,
+)
+
+# this would be anext(obj) if it existed
+anext_op = custom_op(
```
Comment (ilevkivskyi) [inline mypyc/primitives/generic_ops.py:347]: Since you are adding primitives, maybe also add few IR tests?

**Review by ilevkivskyi — APPROVED** [review]

File: mypyc/primitives/generic_ops.py (hunk 1 above)
Comment (msullivan) [reply mypyc/primitives/generic_ops.py:347]: I'm not sure if it would add much value here?

Comment (msullivan) [reply mypyc/primitives/generic_ops.py:347]: (I went and merged but will add IR tests if you think it's important)

Comment (ilevkivskyi) [reply mypyc/primitives/generic_ops.py:347]: They are much faster than run tests, and also I remember similar tests for non-async for were helpful when I implemented some special-casing a while ago (but maybe I care too much about generated code looking good, not just being correct).

**Pull request merged**